flash/core/data/data_source.py

```diff
@@ -461,3 +470,68 @@ class TensorDataSource(SequenceDataSource[torch.Tensor]):
 class NumpyDataSource(SequenceDataSource[np.ndarray]):
     """The ``NumpyDataSource`` is a ``SequenceDataSource`` which expects the input to
     :meth:`~flash.core.data.data_source.DataSource.load_data` to be a sequence of ``np.ndarray`` objects."""
+
+
+class FiftyOneDataSource(DataSource[SampleCollection]):
+    """The ``FiftyOneDataSource`` expects the input to
+    :meth:`~flash.core.data.data_source.DataSource.load_data` to be a ``fiftyone.core.collections.SampleCollection``."""
```

Comment:
Is fiftyone using sphinx? If so, we could add it to our [intersphinx mapping](https://github.com/PyTorchLightning/lightning-flash/blob/447c28ee39139f5424f3726b00eb6a500394ea19/docs/source/conf.py#L88) to have these as`:class:` style links.

Reply:
We do use sphinx, thanks for pointing out intersphinx mapping, we'll go ahead and add that to our repository as well!